src/Workspaces/Core/Portable/Editing/DeclarationKind.cs

```diff
@@ -37,10 +46,18 @@ public enum DeclarationKind
         Attribute,
         LambdaExpression,
         GetAccessor,
+
+        /// <summary>
+        /// Represents set accessor declaration of a property, including init accessors in C#.
+        /// </summary>
         SetAccessor,
+
         AddAccessor,
         RemoveAccessor,
         RaiseAccessor,
+
+        [Obsolete($"This value is not used. Use {nameof(Class)} instead.")]
+        [EditorBrowsable(EditorBrowsableState.Never)]
         RecordClass,
```

Comment:
:warning: This enum member was shipped, but it's still in Unshipped text file. I suspect this is unintentional?

Reply:
@333fred

Reply:
This case was missed during the 3.9.0 API review, so we have two options:

1. Fix it for 3.10.0
2. Leave it like it was, and have this be the only special case for this enum

Reply:
I don't understand. The doc comment appears to be wrong, we were using this enum. Why are we obsoleteing it?

Reply:
@333fred The intent of DeclarationKind enum is to represent language-agnostic declarations. For example, it doesn't have the VB modules. The addition of RecordClass didn't follow the intent of the enum. The available options are:

- Keep the inconsistency, and start adding language-specific declarations, e.g, VB modules, C# init accessors, record classes, and record structs.
- Delete RecordStructs since it's not shipped, and obsolete (and don't use) RecordClasses.

@sharwell and @CyrusNajmabadi preferred the second option.